altlas/Assets/scripts/MapGenerator.cs

```diff
@@ -1,65 +1,348 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using UnityEngine;
 
 public class MapGenerator : MonoBehaviour {
 
     public Transform map;
-    private float DRAWER_HOR_LEN = 0.36f;
-    private float DRAWER_VER_LEN = 0.44f;
+    private Loader loader;
+    private bool once = true;
 
-    List<MapClass> allMaps = new List<MapClass>();
+    //SCENE CONSTANTS FOR PLACING
+    private float DRAWER_HOR_LEN = 0.12f;
+    private float DRAWER_VER_LEN = 0.54f;
+    private float DRAWER_SHORT_X_DIS = 0.3f;
+    private float DRAWER_LONG_X_DIS = 1.4f;
+    private float DRAWER_Y_DIS = 0.24f;
 
+    private float DRAWER_Z_POSITION = 0.37f;
+    private float STARTING_DRAWER_X_POSITION = 0.70f;
+    private float STARTING_DRAWER_Y_POSITION = 0.6f;
+    private float MAPS_Y_OFFSET = 0.01f;
+
+    private Vector3 DESK_FREE_AREA_LEFT_CORNER = new Vector3(-0.7105434f, 0.85f, 0.4124395f);
+    private float DESK_FREE_AREA_LENGTH = 0.4f;
+
+    //List<MapData> allMaps = new List<MapData>();
+
+    Dictionary<string, Dictionary<string, List<MapData>>> subCats = new Dictionary<string, Dictionary<string, List<MapData>>>();
     // Use this for initialization
-    void Start () {
-        
+    void Start() {
+        loader = GetComponent<Loader>();
+
+        // TESTING //
+        MapData map1 = new MapData(1960, "language", "coordinate", "title", "source", new int[] { 21, 21 }, "property", "description", "astronomie", "astronomie", "location", "HK 1305");
```

Comment:
Can we remove testing code or move it to a different file so we keep our code clean